src/views/auths/2FaForm.jsx

```diff
@@ -0,0 +1,107 @@
+    setOtpfileds(newOTP);
+    // checking whether there is a value in field
+    if (!value) setActiveField(index - 1);
+    else setActiveField(index + 1);
+  };
+  //  handle focus every time input needed to the next field
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [activeField]);
+  // handle submition of OTP for verification
+  const handleVerify = async () => {
+    // eslint-disable-next-line no-alert
+    setverifyButton({
+      display: 'inline-block',
+      text: `Loading...`,
+    });
+    const OTP = otpFields.join('');
+    await fetch(`${URL}/api/vendor/verify`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ verificationCode: OTP }),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.token) {
+          toast.success('here you are vendor!', { theme: 'colored' });
+          localStorage.setItem('token', JSON.stringify(data.token));
+          navigate('/vendor-dashboard');
+        } else if (OTP.length !== 6) {
+          toast.error('All digits must be completed ', { theme: 'colored' });
+        } else {
+          toast.error(data.message, { theme: 'colored' });
+        }
+      })
+      .catch((err) => {
+        toast.error(err.message, { theme: 'colored' });
+      });
+    setverifyButton({
+      display: 'hidden',
+      text: `verify OTP`,
+    });
+  };
+  return (
+    <div className="bg-darkGrey h-[100vh] m-0 p-0 flex items-center justify-center w-full overflow-hidden relative">
+      {/* <ToastContainer theme="colored" /> */}
+      <div className="md:w-full md:rounded-none sm:w-3/3 w-2/3 h-[110vh] bg-darkBlueColor absolute right-[-20%] md:right-0 rounded-full rounded-r overflow-hidden" />
+      <div className="px-8 pt-[10vh] -mt-10 bg-whiteColor w-2/4 h-[50vh] md:h-3/4 backdrop-blur-md flex flex-col items-center rounded-md shadow-2xl md:w-[90%] md:rounded-r sm:px-2 sm:w-[98%] md:px-10">
+        <h1 className="h-fit text-2xl font-semibold mb-2 text-center capitalize font-light">
+          welcome back, vendor!
+        </h1>
+        <span className="text-md text-center capitalize font-thin">
+          verify your OTP to get in.
+        </span>
+        <div className="flex flex-row justify-around mt-10">
+          {otpFields.map((_, index) => {
+            return (
+              // eslint-disable-next-line react/no-array-index-key
+              <React.Fragment key={index}>
+                <OTPinput
+                  styling="m-2 border border-darkBlueColor h-10 w-10 text-center form-control rounded placeholder-gray-400 focus:placeholder-gray-200 number-ring"
+                  handleChange={handleChange}
+                  index={index}
+                  inputRef={inputRef}
+                  active={activeField}
+                  value={otpFields}
+                />
+              </React.Fragment>
+            );
+          })}
```

Comment:
Let's create a custom component for this. We should create re-usable components whenever possible.